### DISCUSSION TITLE
Server-side option in kubectl apply

### DISCUSSION THREAD

**Discussion by robervexcel** [helmfile/helmfile#1001] [Ideas]
Hi all. I have a suggestion about this option because in Keda helm chart needs a` --server-side` option to deploy the last version of the chart [https://github.com/kedacore/keda/issues/4740](url) because the CRD is too long 
`The CustomResourceDefinition "scaledjobs.keda.sh" is invalid: metadata.annotations: Too long: must have at most 262144 bytes`

 I searched in the documentation and I tried with options:
 
 `  disableValidation: false
    disableValidationOnInstall: false
    disableOpenAPIValidation: false`
 
  but I didn't find an alternative
  
  more info: [https://kubernetes.io/docs/reference/using-api/server-side-apply/](https://kubernetes.io/docs/reference/using-api/server-side-apply/)

**Comment by yxxhero** [comment]
@robervexcel please show your helmfile.

**Reply by robervexcel** [reply]
<code>
filepath: ""
namespace: keda
repositories:
- name: kedacore
  url: https://kedacore.github.io/charts
releases:
- chart: kedacore/keda
  version: 2.11.2
  name: keda
  labels:
    values.jenkins-x.io: lock
  values:
  - keda.yaml.gotmpl
templates: {}
renderedvalues: {}
</code>

**Comment by lucasfcnunes** [comment]
Isn't it `server-side`?

**Reply by robervexcel** [reply]
Ouch, yes, fail to me, updated the title and description.

**Comment by robervexcel** [comment]
We have another chart with the same "problem" https://github.com/grafana/agent/issues/4379

`The CustomResourceDefinition "grafanaagents.monitoring.grafana.com" is invalid: metadata.annotations: Too long: must have at most 262144 bytes`
the solution at the moment is still the `server-side` option

**Reply by yxxhero** [reply]
@robervexcel maybe you want a flag for server-side?

**Reply by robervexcel** [reply]
yep I think that is a workaround to solve the problems with this kind of charts with annotations too large
